src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,711 @@
+   * @param[out] pt2LongestSegment1 the second point of one of the longest segment.
+   * @param[out] pt3LongestSegment2 the first point of one of the second longest segment.
+   * @param[out] pt4LongestSegment2 the second point of one of the second longest segment.
+   *
+   * @note the segment bounding box can be drawn with the sequence of
+   * out parameters pt1LongestSegment1, pt2LongestSegment1,
+   * pt3LongestSegment1, pt4LongestSegment1.
+   **/
+  void getBoundingBox(PointD &pt1LongestSegment1,
+                      PointD &pt2LongestSegment1,
+                      PointD &pt3LongestSegment2,
+                      PointD &pt4LongestSegment2) const;
+  
+    
+  
+  
+
+   /**
+     * @return the style name used for drawing this object.
+     */
+    std::string className() const;
+
+
+
+  
+  /**
+   * Writes/Displays the object on an output stream.
+   *
+   * @param[out] out the output stream where the object is written.
+   */
+  void selfDisplay ( std::ostream & out ) const;
+  
+    // ------------------------- Protected Datas ------------------------------
+protected:
+  
+  
+  
+  /**
+   * begin iterator (associated to input data)
+   **/
+  ConstIterator myBegin;
+  
+  /**
+   * begin iterator (associated to input data)
+   **/
+  ConstIterator myEnd;
+  
+    
+
+
+    // ------------------------- Private Datas --------------------------------
+private:
+
+  /**
+   * The set of points contained in the alpha thick segment which can be changed during computations.
+   **/
+  mutable InputPointContainer myPointContainer; 
+
+  /**
+   * The maximal thickness of the segment.
+   */
+  double myMaximalThickness;  
+
+  /**
+   * State of the actual computer
+   **/ 
+  State myState;
+
+  /**
+   * Previous saved computer state
+   **/   
+  mutable State _state;
+
+  
+  bool myIsStoringPoints;
+  
+  /**
+   * Used by the size method.
+   **/
+  unsigned int myNbPointsAddedFromIterators;
+
+
+    // ------------------------- Hidden services ------------------------------
+protected:
+
+  
+  /**
+   *  Used to check if the initialisation with 3 non aligned points is well done.
+   *
+   * @return 'true' if the segment contains at least 3 non aligned points. 
+   **/     
+   bool melkmanIsWellInitialized() const;
+
+
+  /**
+   *  Updates the main height of the melkman convex set and return the new thickness of the segment.
+   * 
+   * @return the thickness of the segment. 
+   **/     
+  double melkmanUpdateMainHeight();
+  
+  
+  /**
+   * Adds a point in the convex set  using one step of Melkman algorithm.
+   *
+   * @param[in] aPoint the point to be added.
+   */
+  void melkmanAddPoint(const InputPoint  &aPoint );
+
+
+
+  /**
+   * Init melkman algorithm with the main queue and the 3 input first points.
+   **/
+  void melkmanInit( ) ; 
+
+
+  /**
+   * Depending on connexity, return true if a convex is valid.
+   */
+  bool melkmanIsConvexValid() ;
+  
+  
+  /**
```

Comment:
Should the Melkman algorithm be implementer elsewhere as a generic tool ? In geometry/curves or geometry/helpers ?